libbeat/autodiscover/providers/docker/docker.go

```diff
@@ -119,8 +119,12 @@ func (d *Provider) emitContainer(event bus.Event, flag string) {
 	}
 
 	labelMap := common.MapStr{}
+	labels := common.MapStr{}
```

Comment:
is this a `map[string]string`? Perhaps that would make it more obvious that it is not used differently?

Reply:
we have standardized on `MapStr` for labels/annotations/hints on all of the autodiscover related code. I kept it the same for consistency sake. @exekias thoughts?

Reply:
I'm wondering if we should change `GenerateHints` instead, to take MapStr nesting into account. We are enforcing MapStr nesting to avoid mapping issues in Elasticsearch, so this will be a pattern. Having this in the "unsafe" way may lead to different results, what do you think?